### PR TITLE
DSY-1948 refactor list

### PR DESCRIPTION
**Pull request by arielwb**
# Description

Please include a summary of the changes and features that were added. Also include which JIRA task is this related to (name and link). List any dependencies that were added or removed for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/components/List/stories.data.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    key: 'item3',
+    title: 'A Storm of Swords',
+  },
+  {
+    key: 'item4',
+    title: 'A Feast for Crows',
+  },
+  {
+    extraInfo: {
+      author: 'George R.R. Marting',
```
Comment (lauralucca) [inline src/components/List/stories.data.ts:20]: typo: Martin

File: src/components/List/List.stories.tsx [hunk 2]
```diff
@@ -1,237 +1,138 @@
...
+import { ListItem, ListItemFeedback } from './ListItem';
+import { Checkbox } from '../Checkbox';
+import { TextWithTheme } from '../../common/HelperComponents/ThemeHelper.styles';
+import { data } from './stories.data';
 
 export default {
-  component: List,
+  component: ListItem,
   parameters: {
     componentSubtitle: 'List let users choose between several items',
```
Comment (lauralucca) [inline src/components/List/List.stories.tsx:14]: também tem as listas não selecionáveis né

File: src/components/ListItem/ListItem.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+import { Divider } from '../Divider';
+import { ListItem, ListItemFeedback } from './ListItem';
+import { Checkbox } from '../Checkbox';
+import { TextWithTheme } from '../../common/HelperComponents/ThemeHelper.styles';
+import { data } from './stories.data';
+
+export default {
+  component: ListItem,
+  parameters: {
+    componentSubtitle: 'List items are rows information that can have user interactions or not',
```
Comment (lauralucca) [inline src/components/ListItem/ListItem.stories.tsx:14]: rows of information? desculpa a chatice, documentaçao né ><

File: src/components/ListItem/ListItem.styles.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+} from '../../common/themeSelectors';
+import { ListItemProps } from './ListItem';
+
+const backgroundStyles = ({ selected, theme }: Required<ListItemProps>) => (
+  selected
+    ? buildColorWithOpacity(getColorPrimary, getOpacityMediumLow, theme)
+    : 'transparent'
+);
+
+const getSizeMedium = getTheme(theme => getSize(theme, 'medium'));
```
Comment (lauralucca) [inline src/components/ListItem/ListItem.styles.ts:17]: não entendi por que isso tá aqui. é seletor do tema?

File: test/testHelpers.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  ui,
+  { wrapper: WithTheme, ...options },
+);
+
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+export * from '@testing-library/react-native';
+
+export {
+  customRender as render,
```
Comment (lauralucca) [inline test/testHelpers.tsx:26]: Lendo os testes eu fiquei pensando se nao vale a pena deixar `customRender` ou `renderWithTheme` pra não confundir com o render da testing library caso uma pessoa leia com menos atençao

File: src/components/ListItem/ListItem.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+export type ListItemFeedback = 'ripple' | 'selection'
+
+export interface ListItemProps {
+  /**
+   * The children list item elements
+   */
+  children: ReactElement;
+  /**
+   * Controls the element feedback type
```
Comment (lauralucca) [inline src/components/ListItem/ListItem.tsx:15]: põe aqui que as opções são 'ripple' e 'selection'? pra aparecer na doc que não apareceu

File: src/components/ListItem/ListItem.styles.ts (hunk 4 above)
Comment (arielwb) [reply src/components/ListItem/ListItem.styles.ts:17]: não é um seletor, é so uma funcão que usa dois seletores, (getSize e getTheme).
sem isso, o codigo ficaria assim

`min-height: ${({theme}) =>getSizeMedium(theme, 'medium')}`

File: src/components/ListItem/ListItem.stories.tsx (hunk 3 above)
Comment (arielwb) [reply src/components/ListItem/ListItem.stories.tsx:14]: não entendi a duvida

Comment (arielwb) [reply src/components/ListItem/ListItem.stories.tsx:14]: mas se a duvida é da escolha das palavras, eu tirei daqui: https://developer.apple.com/design/human-interface-guidelines/watchos/elements/lists-and-tables/

File: src/components/ListItem/ListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+   * The children list item elements
+   */
+  children: ReactElement;
+  /**
+   * Controls the element feedback type `ripple` | `selection`
+   * @Default ripple
+   */
+  feedback?: ListItemFeedback;
+  /**
+   * Controls the element selected styles feedback type is selection
```
Comment (lauralucca) [inline src/components/ListItem/ListItem.tsx:20]: Não entendi essa frase

File: src/components/ListItem/ListItem.stories.tsx (hunk 3 above)
Comment (lauralucca) [reply src/components/ListItem/ListItem.stories.tsx:14]: é que faltou o OF

Comment (arielwb) [reply src/components/ListItem/ListItem.stories.tsx:14]: AAAAAAH

**Review by lauralucca — APPROVED** [review]
LGTM

**Pull request merged**